Live-Snap/Live-Snap/Controllers/SelectPhotoViewController.swift

```diff
@@ -70,15 +73,46 @@ class SelectPhotoViewController: UIViewController, UICollectionViewDataSource, U
         progressView = UIView()
         activityIndicator = UIActivityIndicatorView()
         
+        deniedLibraryPermissionLabel = UILabel(frame: CGRect(x: 0, y: 0, width: view.frame.width * 0.75, height: 100))
+        deniedLibraryPermissionLabel.center = view.center
+        deniedLibraryPermissionLabel.textColor = UIColor.snapWhite
+        deniedLibraryPermissionLabel.textAlignment = .center
+        deniedLibraryPermissionLabel.text = "Live-Snap needs access to your photo library to import and export images. Please click the settings button below and enable access"
+        deniedLibraryPermissionLabel.numberOfLines = 4
+        
+        deniedLibraryPermissionButton = UIButton(frame: CGRect(x: 0, y: view.frame.height - 50, width: view.frame.width, height: 50))
+        deniedLibraryPermissionButton.setTitle("Settings", for: .normal)
+        deniedLibraryPermissionButton.setTitleColor(UIColor.snapBlack, for: .normal)
+        deniedLibraryPermissionButton.setBackgroundImage(UIImage(color: UIColor.snapYellow, size: deniedLibraryPermissionButton.frame.size), for: .normal)
+        deniedLibraryPermissionButton.setBackgroundImage(UIImage(color: UIColor.snapYellow.withAlphaComponent(0.75), size: deniedLibraryPermissionButton.frame.size), for: .highlighted)
+        deniedLibraryPermissionButton.addTarget(self, action: #selector(settingsButtonWasPressed), for: .touchUpInside)
+        
         statusBarView = UIView(frame: UIApplication.shared.statusBarFrame)
         statusBarView.backgroundColor = UIColor.snapBlack
         
-        view.addSubview(collectionView)
-        view.addSubview(tableView)
-        view.addSubview(toolBar)
-        view.addSubview(statusBarView)
+        ImageManager.shared.requestPhotoLibraryPermission { (status) in
+            DispatchQueue.main.async {
+                if self.deniedLibraryPermissionButton.isDescendant(of: self.view) {
+                    self.deniedLibraryPermissionButton.removeFromSuperview()
+                }
+                
+                if self.deniedLibraryPermissionLabel.isDescendant(of: self.view) {
+                    self.deniedLibraryPermissionLabel.removeFromSuperview()
+                }
+                
+                if status == false {
```

Comment:
have you tried running it, denying permission, closing the app and running it again?